src/coreclr/jit/gentree.cpp

```diff
@@ -6566,7 +6566,7 @@ ExceptionSetFlags GenTree::OperExceptions(Compiler* comp)
                 return ExceptionSetFlags::NullReferenceException;
             }
 
-            return ExceptionSetFlags::None;
+            return ExceptionSetFlags::IndexOutOfRangeException;
```

Comment:
The case above also needs this I assume?

Reply:
Yes... it is getting late in the day.

Technically `ARR_INDEX` also needs it. Left it alone for now, we presumably expect to delete it.